include/inexor/vulkan-renderer/wrapper/device.hpp

```diff
@@ -7,80 +7,95 @@
+    /// ones is suitable), an exception is thrown! It is the responsibility of the caller to handle this!
+    /// @param inst The Vulkan instance
     /// @param surface The window surface
-    /// @return ``true`` if presentation is supported
-    [[nodiscard]] static bool is_presentation_supported(VkPhysicalDevice graphics_card, VkSurfaceKHR surface);
-
-    /// @brief Default constructor.
-    /// @param instance The instance wrapper from which the device will be created
-    /// @param surface The surface which will be associated with the device
-    /// @param enable_vulkan_debug_markers ``true`` if Vulkan debug markers should be enabled
-    /// @param prefer_distinct_transfer_queue ``true`` if a distinct data transfer queue should be preferred
-    /// @param preferred_physical_device_index The index of the preferred graphics card which should be used,
-    /// starting from 0. If the graphics card index is invalid or if the graphics card is unsuitable for the
-    /// application's purpose, another graphics card will be selected automatically. See the details of the device
-    /// selection mechanism.
-    /// TODO: Add overloaded constructors for VkPhysicalDeviceFeatures and requested device extensions
-    Device(const wrapper::Instance &instance, VkSurfaceKHR surface, bool enable_vulkan_debug_markers,
-           bool prefer_distinct_transfer_queue,
-           std::optional<std::uint32_t> preferred_physical_device_index = std::nullopt);
+    /// @param required_features The required device features
+    /// @param required_extensions The required device extensions
+    /// @return The chosen physical device which is most suitable
+    static VkPhysicalDevice pick_best_physical_device(const Instance &inst, VkSurfaceKHR surface,
+                                                      const VkPhysicalDeviceFeatures &required_features,
+                                                      const std::vector<const char *> &required_extensions);
+
+    /// Default constructor
+    /// @warning This constructor can throw exceptions! It is the responsibility of the caller to handle this!
+    /// @param inst The Vulkan instance
+    /// @param surface The window surface
+    /// @param prefer_distinct_transfer_queue Specifies if a distinct transfer queue will be preferred
+    /// @param required_features The required device features which the physical device must all support
+    /// @param optional_features The optional device features which do not necessarily have to be present
+    /// @note The creation of the physical device will not fail if one of the optional device features is not available
+    Device(const Instance &inst, VkSurfaceKHR surface, bool prefer_distinct_transfer_queue,
+           VkPhysicalDevice physical_device, const VkPhysicalDeviceFeatures &required_features,
+           const std::vector<const char *> &required_extensions,
+           const VkPhysicalDeviceFeatures &optional_features = {});
```

Comment:
```suggestion
    Device(const Instance &inst, VkSurfaceKHR surface, bool prefer_distinct_transfer_queue,
           VkPhysicalDevice physical_device, const std::vector<const char *> &required_extensions,
           const VkPhysicalDeviceFeatures &required_features,
           const VkPhysicalDeviceFeatures &optional_features = {});
```

Reply:
I also completely forgot about the documentation of `required_extensions` and `physical_device`. Done.